### ISSUE TITLE
Anvil crashes with `index out of bounds` when requesting `debug_traceTransaction`

### ISSUE THREAD

**Issue by michprev** [foundry-rs/foundry#5971]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (8e64cff 2023-10-01T00:20:28.991654440Z)

### What command(s) is the bug in?

anvil --prune-history 100 --transaction-block-keeper 10 --steps-tracing --silent --port 52385

### Operating System

Linux

### Describe the bug

Anvil crashes with the following error message when requesting `debug_traceTransaction` for a failed tx:

```
thread 'tokio-runtime-worker' panicked at 'index out of bounds: the len is 0 but the index is 0', crates/evm/src/trace/mod.rs:141:45
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I have created a reproducible example here: https://github.com/michprev/anvil-low-gas-debugTT-crash

The issue probably is related to the fact that a contract is being created in a call that does not have enough gas (see https://github.com/michprev/anvil-low-gas-debugTT-crash/blob/main/contracts/Foo.sol).

**Comment by mattsse** [comment]
most of these have been fixed in the reth tracer impl, I'm hoping to upstream this to anvil very shortly

**Comment by Evalir** [comment]
Will be closed by #6219, pre-emptively closing
